### PR TITLE
Workaround for qemu >= 2.1.1 - tested on qemu 3.1

### PR DESCRIPTION
**Pull request by zi0Black**
Probably not enough to be a real fix but it is enough to be a workaround that is allowing me to get a working emulation of a mipseb firmware of the `Netgear WNAP320`, with host system ubuntu 19 and qemu `qemu (1:3.1+dfsg-2ubuntu3.7) disco`.

rev-shell-qemu-netgear

### CODE REVIEW

File: scripts/makeNetwork.py [hunk 1]
```diff
@@ -138,7 +138,8 @@ def qemuArchNetworkConfig(i, arch, n):
         if arch == "arm":
             return "-device virtio-net-device,netdev=net%(I)i -netdev socket,id=net%(I)i,listen=:200%(I)i" % {'I': i}
         else:
-            return "-net nic,vlan=%(VLAN)i -net socket,vlan=%(VLAN)i,listen=:200%(I)i" % {'I': i, 'VLAN' : i}
+            #workaround for qemu >= 2.1.1 tested on Ubuntu 19 with qemu 3.1
+            return "-netdev user,id=n%(VLAN)i -device e1000,netdev=n%(VLAN)i -nic socket,listen=:200%(I)i" % {'I': i, 'VLAN' : i}
```
Comment (ddcc) [inline scripts/makeNetwork.py]: `%(VLAN)i` should be changed to `%(I)i`, and you can drop the `VLAN` field from the dictionary.

File: scripts/makeNetwork.py [hunk 2]
```diff
@@ -147,7 +148,8 @@ def qemuArchNetworkConfig(i, arch, n):
         else:
             vlan_id = vlan if vlan else i
             mac_str = "" if not mac else ",macaddr=%s" % mac
```
Comment (ddcc) [inline scripts/makeNetwork.py]: The `",macaddr=%s"` part should be changed to ",mac=%s'

File: scripts/makeNetwork.py [hunk 3]
```diff
@@ -147,7 +148,8 @@ def qemuArchNetworkConfig(i, arch, n):
...
-            return "-net nic,vlan=%(VLAN)i%(MAC)s -net tap,vlan=%(VLAN)i,id=net%(I)i,ifname=${TAPDEV_%(I)i},script=no" % { 'I' : i, 'MAC' : mac_str, 'VLAN' : vlan_id}
+            #workaround for qemu >= 2.1.1 tested on Ubuntu 19 with qemu 3.1
+            return "-netdev tap,id=net%(I)i,ifname=${TAPDEV_%(I)i},script=no -device e1000,netdev=net%(I)i" % { 'I' : i, 'MAC' : mac_str, 'VLAN' : vlan_id}
```
Comment (ddcc) [inline scripts/makeNetwork.py]: You can drop the `VLAN` field as well, but you need to append the MAC address field to the end of the `device` argument, like `netdev=net%(I)i%(MAC)s`.

**Comment by ddcc** [comment]
Thanks for making these changes! There're just a few more tweaks to make sure this doesn't break other existing functionality.

**Comment by zi0Black** [comment]
Okay, I've made the necessary adjustments, see if everything's okay now.

File: scripts/makeNetwork.py (hunk 2 above)
Comment (zi0Black) [reply scripts/makeNetwork.py]: Fixed  in the commit `bf57905`.

File: scripts/makeNetwork.py (hunk 3 above)
Comment (zi0Black) [reply scripts/makeNetwork.py]: Fixed  in the commit `bf57905`.

File: scripts/makeNetwork.py (hunk 1 above)
Comment (zi0Black) [reply scripts/makeNetwork.py]: Fixed  in the commit `bf57905`.

**Comment by ddcc** [comment]
Thanks, looks great! One last change, could you also remove the following text in README.md?

```**Only QEMU <= 2.12 is supported! See bugs [#96](https://github.com/firmadyne/firmadyne/issues/96) and [#119](https://github.com/firmadyne/firmadyne/issues/119) for more information.**```

**Comment by zi0Black** [comment]
Okay, I have update the readme.

**Pull request merged**